### PR TITLE
safe logout rpc (CORE-10775, DESKTOP-9718)

### PR DESCRIPTION
**Pull request by heronhaye**

### CODE REVIEW

File: go/service/login.go [hunk 1]
```diff
@@ -28,8 +30,21 @@ func (h *LoginHandler) GetConfiguredAccounts(context context.Context, sessionID
...
+func (h *LoginHandler) Logout(ctx context.Context, arg keybase1.LogoutArg) (err error) {
 	defer h.G().CTraceTimed(ctx, "Logout [service RPC]", func() error { return err })()
+
+	if !arg.ForceDangerous {
+		res, err := libkb.CanLogout(libkb.NewMetaContext(ctx, h.G()))
+		if err != nil {
+			return err
+		}
+		if !res.CanLogout {
+			return fmt.Errorf("cannot logout: %s; try with force", res.Reason)
```
Comment (zapu) [inline go/service/login.go]: I'm not sure about this error message especially if it ends up in GUI.

Even in cli, I don't think encouraging users to just `--force`  it is a good idea.

**Comment by zapu** [comment]
Needs frontend change because function signature has changed:
```
actions/config/index.js:271:12 Cannot call `Saga.callPromise` because function type [1] requires another argument from function type [2]. actions/config/index.js:271:12 271
```

**Comment by heronhaye** [comment]
Yeah, I tagged cjb on the ticket

**Comment by zapu** [comment]
@heronhaye @cjb should we try to merge this?

**Comment by heronhaye** [comment]
@cjb this PR is still open if you wanted to get it in. or we can close it if not

**Comment by heronhaye** [comment]
killing this pr

**Comment by heronhaye** [comment]
killing this pr

**Pull request closed without merging**